collectors/python.d.plugin/openldap/openldap.chart.py

```diff
@@ -0,0 +1,200 @@
+            ['entries_sent', 'sent', 'incremental']
+        ]
+    },
+    'ldap_operations': {
+        'options': [None, 'Operations', 'ops/s', 'ldap', 'openldap.ldap_operations', 'line'],
+        'lines': [
+            ['bind_operations', 'bind', 'incremental'],
+            ['search_operations', 'search', 'incremental'],
+            ['unbind_operations', 'unbind', 'incremental'],
+            ['add_operations', 'add', 'incremental'],
+            ['delete_operations', 'delete', 'incremental'],
+            ['modify_operations', 'modify', 'incremental'],
+            ['compare_operations', 'compare', 'incremental']
+        ]
+    },
+    'waiters': {
+        'options': [None, 'Waiters', 'waiters/s', 'ldap', 'openldap.waiters', 'line'],
+        'lines': [
+            ['write_waiters', 'write', 'incremental'],
+            ['read_waiters', 'read', 'incremental']
+        ]
+    },
+}
+
+# Stuff to gather - make tuples of DN dn and attrib to get
+SEARCH_LIST = {
+    'total_connections': (
+        'cn=Total,cn=Connections,cn=Monitor', 'monitorCounter',
+    ),
+    'bytes_sent': (
+        'cn=Bytes,cn=Statistics,cn=Monitor', 'monitorCounter',
+    ),
+    'completed_operations': (
+        'cn=Operations,cn=Monitor', 'monitorOpCompleted',
+    ),
+    'initiated_operations': (
+        'cn=Operations,cn=Monitor', 'monitorOpInitiated',
+    ),
+    'referrals_sent': (
+        'cn=Referrals,cn=Statistics,cn=Monitor', 'monitorCounter',
+    ),
+    'entries_sent': (
+        'cn=Entries,cn=Statistics,cn=Monitor', 'monitorCounter',
+    ),
+    'bind_operations': (
+        'cn=Bind,cn=Operations,cn=Monitor', 'monitorOpCompleted',
+    ),
+    'unbind_operations': (
+        'cn=Unbind,cn=Operations,cn=Monitor', 'monitorOpCompleted',
+    ),
+    'add_operations': (
+        'cn=Add,cn=Operations,cn=Monitor', 'monitorOpInitiated',
+    ),
+    'delete_operations':  (
+        'cn=Delete,cn=Operations,cn=Monitor', 'monitorOpCompleted',
+    ),
+    'modify_operations': (
+        'cn=Modify,cn=Operations,cn=Monitor', 'monitorOpCompleted',
+    ),
+    'compare_operations': (
+        'cn=Compare,cn=Operations,cn=Monitor', 'monitorOpCompleted',
+    ),
+    'search_operations': (
+        'cn=Search,cn=Operations,cn=Monitor', 'monitorOpCompleted',
+    ),
+    'write_waiters': (
+        'cn=Write,cn=Waiters,cn=Monitor', 'monitorCounter',
+    ),
+    'read_waiters': (
+        'cn=Read,cn=Waiters,cn=Monitor', 'monitorCounter',
+    ),
+}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+
+        self.server = configuration.get('server')
+        self.port = configuration.get('port')
+        self.username = configuration.get('username')
+        self.password = configuration.get('password')
+        self.timeout = configuration.get('timeout')
```

Comment:
`configuration.get('timeout', 1)`

i suggest `1` as default

Reply:
add global var `DEFAULT_TIMEOUT` and use it in `configuration.get`

Reply:
Sorry I don't realy get you. I already added a new variable inside `openldap.conf` where I think we store all plugin "settings" variables. I set it there with value `1`. If I create a new `DEFAULT_TIMEOUT` globaly how will I use it with configuration.get. "configuration" method is loaded inside service class. Anyway I added a default value as you suggested.

Reply:
i meant this


```python
DEFAULT_TIMEOUT = 1

ORDER = [
...
...
...

self.timeout = configuration.get('timeout', DEFAULT_TIMEOUT)
```

Reply:
done.